### ISSUE TITLE
Feature - Don't allow for a fake to be called directly

### ISSUE THREAD

**Issue by tgandrews** [sinonjs/sinon#1592]
* Sinon version : 3.3.0
* Environment  : Node 8.7.0
* Other libraries you are using: Chai & Jest

**What did you expect to happen?**
```
const stub = require('sinon').stub;
const thingStub = stub().withArgs(1).returns('a');
console.log(thingStub(2)); // Expect undefined but found 'a'
```
**What actually happens**
It runs as if the `withArgs` was never applied as it invokes the `fake` returned. This I found very confusing and originally thought it was a bug. I understand it is like this because the stub keeps a reference to the fakes and the fakes are configured to respond in particular ways. `withArgs` needs to return a fake so it can be configured by the `returns` call.

**Request**
To return some sort of error (if possible) if a fake is invoked directly. 

**Reasoning**
Basically I have a code base where this error silently lay in the code base until it was picked up by code coverage of all things. As particular paths weren't being covered as they weren't being run as the expected stub value wasn't coming back. I appreciate better testing practice would have caught this but I found this bug quite subtle and did not see anything obvious in the documentation.

Any suggestions on the change location and/or documentation would be appreciated and I would be more than happy to make the change.
